BDArmory/Control/MissileFire.cs

```diff
@@ -6222,7 +6222,7 @@ public void SendTargetDataToMissile(MissileBase ml, bool clearHeat = true)
                         Debug.Log("[BDArmory.MissileData]: Sending targetInfo to laser Missile...");
                     if (guardMode && ((foundCam.groundTargetPosition - guardTarget.CoM).sqrMagnitude < 10 * 10))
                     {
-                        ml.targetVessel = guardTarget.gameObject.GetComponent<TargetInfo>();
+                        ml.targetVessel = guardTarget.gameObject ? guardTarget.gameObject.GetComponent<TargetInfo>() : null;
```

Comment:
Shouldn't these also check that `guardTarget` is not `null`?
I see for radar targets and for dropping bombs that this check is made, but not for the other target types.

Reply:
There also seems to be a fair amount of duplicate code here. Much of this could be condensced into a `switch` statement with multiple cases handled by the same block of code.

Reply:
Took a look, and managed to collate and condense the dupe stuff a bit, but using a switch statement doesn't look like it'd reduce the duplicate stuff overmuch? as there's enough specific checks and such that it's turn into a switch version of what's currently dune via ``else if``. Could be I'm just looking at this from the wrong perspective?

Reply:
OK, well a `switch` statement is still preferable over multiple `else if` clauses when possible for performance reasons.